src/NetworkMap/Interfaces/Update.elm

```diff
@@ -0,0 +1,125 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+-}
+
+import Array exposing (Array)
+import Dict exposing (Dict)
+import Maybe exposing (andThen)
+import NetworkMap.Interfaces.Model exposing (Arg, Function, Inputs, Model, Module)
+import NetworkMap.Interfaces.Msg exposing (Msg(..))
+import NetworkMap.Interfaces.Port as Port
+
+
+getArgs : String -> String -> Inputs -> Array Arg
+getArgs moduleName fname inputs =
+    let
+        args =
+            inputs |> Dict.get moduleName |> andThen (Dict.get fname)
+    in
+    Maybe.withDefault Array.empty args
+
+
+update : Msg -> Model -> ( Model, Cmd Msg )
+update msg model =
+    case msg of
+        GetInterface peerId ->
+            ( model, Port.interfacesRequest { command = "get_interface", id = Just peerId, call = Nothing } )
+
+        CallFunction id moduleName fname ->
+            let
+                args =
+                    model.inputs |> getArgs moduleName fname
+
+                argsM =
+                    if Array.isEmpty args then
+                        Nothing
+
+                    else
+                        Just <| Array.toList args
+
+                call =
+                    { moduleName = moduleName, fname = fname, args = argsM }
+            in
+            ( model, Port.interfacesRequest { command = "call", id = Just id, call = Just call } )
+
+        AddInterface interface ->
+            let
+                inputs =
+                    modulesToInputs interface.modules
+            in
+            ( { model | interface = Just interface, inputs = inputs }, Cmd.none )
+
+        UpdateInput moduleId functionId idx input ->
+            let
+                updated =
+                    model.inputs |> Dict.update moduleId (moduleUpdate input functionId idx)
+            in
+            ( { model | inputs = updated }, Cmd.none )
+
+        AddResult callResult ->
+            let
+                updated =
+                    model.results |> Dict.update callResult.moduleName (resultUpdate callResult.result callResult.fname)
+            in
+            ( { model | results = updated }, Cmd.none )
+
+        NoOp ->
+            ( model, Cmd.none )
+
+
+resultUpdate : String -> String -> Maybe (Dict String String) -> Maybe (Dict String String)
+resultUpdate value fname old =
+    case old of
+        Maybe.Just o ->
+            Just (o |> Dict.insert fname value)
+
+        Maybe.Nothing ->
+            let
+                newDict =
+                    Dict.empty
+            in
+            Just (newDict |> Dict.insert fname value)
+
+
+modulesToInputs : Dict String Module -> Inputs
+modulesToInputs modules =
+    modules |> Dict.map (\_ -> \m -> functionsToDic m.functions)
+
+
+functionsToDic : Dict String Function -> Dict String (Array String)
```

Comment:
As far as I understand, this function converts functions inside a dict into an array of String values, one per input type.

Am I right? Maybe change the function name to make it more obvious?